### ISSUE TITLE
amp-story: Missing or inaccurate audio description or media alternative

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#32497]
# Missing or inaccurate audio description or media alternative

## WCAG Level
Level A/AA

## Priority
Low

## Pages/screens/components affected
* [Navigation](https://www.gstatic.com/amphtml/stamp/qa/navigation.html)
* [Video](https://www.gstatic.com/amphtml/stamp/qa/video.html)
* [Prerender - Video](https://www.gstatic.com/amphtml/stamp/qa/prerender-video.html?amp_js_v=0.1#origin=https://www.google.com&prerenderSize=1&visibilityState=prerender&paddingTop=0&p2r=0&horizontalScrolling=0&csi=1&aoh=15526927198732&viewerUrl=https://www.google.com&history=1&storage=1&cid=1&cap=swipe,navigateTo,cid,fragment,replaceUrl)
* [AMP Story Player - AMP Story 360](https://www.gstatic.com/amphtml/stamp/qa/amp-story-360.html)

## Description
When a video or synchronised multimedia presentation (with both video and audio content) has important visual information that is not already conveyed by the audio/dialogue, this information needs to be conveyed either through an *audio description* or as a *media alternative*, such as a transcript, to satisfy WCAG 2.1 Level A. For WCAG 2.1 Level AA, an *audio description* must be provided in all cases where video/multimedia content with significant visual information is present.

None of the videos used in the samples provides users with an audio description, nor a media alternative. This issue is marked as "Low" priority, but only because of the sample nature of the videos.

## User impact
Without an accurate audio description (and optionally, a media alternative), people who are blind or have low vision will find it difficult or impossible to understand the content of the video or multimedia presentation.

In the case of the tested samples, the video content is, to a large extent, only a placeholder/representative sample to demonstrate the technical ability to include video. However, in real-world usage, a provision for audio description (with an optional media alternative, such as a transcript) must be made.

## Required solution
Provide audio description, and optionally a media alternative such as a transcript, for every video or synchronised multimedia presentation that has significant visual content. Note that if all visual content is already described as part of the default audio, no additional audio description or media alternative is required.

A transcript must contain all information of the video/presentation - both the visual aspects (on-screen text, descriptions of any relevant actions or images being shown, etc.), and any meaningful audio (narration, speech, meaningful sound effects or music) - in a more traditional (usually text-based) format. A transcript for a video would include the spoken audio, a description of any relevant information and actions taking place in the video, any meaningful on-screen text, as well as any important sound cues, song titles (if the choice of song conveys a particular meaning), and - where relevant - a description of any music used (if the type or mood of the music is meaningful, e.g. '[Suspenseful music]').

An audio description is a real-time narration added to the existing soundtrack, describing important visual details that are not conveyed within the soundtrack itself. Audio descriptions are normally added during pauses in the dialogue.

Typical examples of audio descriptions include the following:

* A video shows a person being interviewed about their hobbies. Each question posed to the interviewee is shown in text as a separate scene in the video, which is followed by the interviewee giving their response. A narrator announces the question text at the same time that it is displayed.
* A page on a furniture website includes an animation showing how to put together a simple bedside table. The existing soundtrack is simply a piece of music. The additional audio description features a narrator describing the process in real time.

Televisions and set-top boxes often include a button that enables audio descriptions for television programmes to be turned on or off. This functionality is also available on certain web-based video platforms – for instance, YouTube allows creators to upload additional audio tracks, and provides an option in its video player for the user to switch between them.

However, not all video platforms allow for the inclusion of multiple audio tracks. As an alternative, there are two approaches you can take:

1. Include the audio description as part of the existing soundtrack. This means that all users will hear the audio description.
2. Create a separate video that includes the additional audio description, and direct users to this separate video from the primary video (for example, through a separate link).

There are also situations where the original video itself simply does not provide sufficient pauses in the audio to allow for additional narration. In these cases, it is generally best to create a separate audio-described version of the video, where the video itself is paused whenever audio description narration is required.

Note: while [WCAG 2.1 Success Criterion 1.2.3 Audio Description or Media Alternative (Prerecorded) (Level A)](https://www.w3.org/TR/WCAG21/#audio-description-or-media-alternative-prerecorded) allows for *either* audio description *or* a media alternative to satisfy WCAG at level A, [WCAG 2.1 Success Criterion 1.2.5 Audio Description (prerecorded) (Level AA)](https://www.w3.org/TR/WCAG21/#audio-description-prerecorded) *only* allows audio description in order to conform with WCAG at level AA.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance

The exact way in which an audio description (with potentially the option to switch between different audio tracks on a video) and media alternative should be implemented will need design/UX and development considerations that are beyond the scope of this evaluation. This could be left up to individual authors to decide - for instance, adding an extra link of page to an AMP story that points to an audio-described version of the video used, and/or to a transcript. Alternatively, it could be something that is designed and developed as part of AMP stories themselves and offered as an additional button/control (similar to the "share" button).

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue.

For all videos or synchronised multimedia presentations:

1. Check that an audio description (either as part of the "regular" audio track, or as a selectable alternative) is available.
2. Check that the audio description provides an accurate representation of the significant visual content in the video/presentation.
3. Check if an optional media alternative (such as a transcript) is available - in the absence of an audio description, this won't help pass WCAG 2.1 Success Criterion 1.2.5 Audio Description (prerecorded) (Level AA), but will at least work as an alternative way of passing WCAG 2.1 Success Criterion 1.2.3 Audio Description or Media Alternative (Prerecorded) (Level A).
4. Check that the optional media alternative provides an accurate representation of the significant visual content in the video/presentation.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.2.3 Audio Description or Media Alternative (Prerecorded) (Level A)](https://www.w3.org/TR/WCAG21/#audio-description-or-media-alternative-prerecorded)
* [WCAG 2.1 Success Criterion 1.2.5 Audio Description (prerecorded) (Level AA)](https://www.w3.org/TR/WCAG21/#audio-description-prerecorded)

## More information
* [WebAIM: Audio Descriptions](https://webaim.org/techniques/captions/#ad)
* [Nomensa: What is audio description?](https://www.nomensa.com/blog/2010/what-audio-description)

## Test data
Test date: January 2021
